pkg/cmd/taskrun/describe.go

```diff
@@ -77,9 +78,18 @@ or
 			}
 
 			if len(args) == 0 {
-				err = askTaskRunName(opts, p)
-				if err != nil {
-					return err
+				if !opts.Last {
+					err = askTaskRunName(opts, p)
+					if err != nil {
+						return err
+					}
+				} else {
+					lOpts := metav1.ListOptions{}
+					trs, err := trlist.GetAllTaskRuns(p, lOpts, opts.Limit)
```

Comment:
You can actually replace opts.Limit with 1 as you just need to return the one TaskRun in this case.

Reply:
Actually Yes, :D correcting it.